tlv/tlv_types_generated.go

```diff
@@ -0,0 +1,1022 @@
+package tlv
+
+import "fmt"
+
+type TlvTypes interface {
+	TlvType0 | TlvType1 | TlvType2 | TlvType3 | TlvType4 | TlvType5 | TlvType6 | TlvType7 | TlvType8 | TlvType9 | TlvType10 | TlvType11 | TlvType12 | TlvType13 | TlvType14 | TlvType15 | TlvType16 | TlvType17 | TlvType18 | TlvType19 | TlvType20 | TlvType21 | TlvType22 | TlvType23 | TlvType24 | TlvType25 | TlvType26 | TlvType27 | TlvType28 | TlvType29 | TlvType30 | TlvType31 | TlvType32 | TlvType33 | TlvType34 | TlvType35 | TlvType36 | TlvType37 | TlvType38 | TlvType39 | TlvType40 | TlvType41 | TlvType42 | TlvType43 | TlvType44 | TlvType45 | TlvType46 | TlvType47 | TlvType48 | TlvType49 | TlvType50 | TlvType51 | TlvType52 | TlvType53 | TlvType54 | TlvType55 | TlvType56 | TlvType57 | TlvType58 | TlvType59 | TlvType60 | TlvType61 | TlvType62 | TlvType63 | TlvType64 | TlvType65 | TlvType66 | TlvType67 | TlvType68 | TlvType69 | TlvType70 | TlvType71 | TlvType72 | TlvType73 | TlvType74 | TlvType75 | TlvType76 | TlvType77 | TlvType78 | TlvType79 | TlvType80 | TlvType81 | TlvType82 | TlvType83 | TlvType84 | TlvType85 | TlvType86 | TlvType87 | TlvType88 | TlvType89 | TlvType90 | TlvType91 | TlvType92 | TlvType93 | TlvType94 | TlvType95 | TlvType96 | TlvType97 | TlvType98 | TlvType99
+}
+
+func TlvTypeStruct(typeVal Type) TlvType {
+	switch typeVal {
+	case 0:
```

Comment:
I'm having a bit of a tough time following what's going on here. Is `Type` an integer?

Reply:
Yeah type is just a unit 8. This can be used to map the actual type value (integer value) to the struct mapping of it. Can prob rename some stuff to make it clearer. 

Also I went down this route as I couldn't use just `TlvType` (the interface) as a type param into `RecordT[T, V]` unless I made it a pointer. So `RecordT[*TlvType1, uint32]` instead of just `RecordT[*TlvType2, uint32]`, maybe that's ok though, was just something I was sorta going after syntax wise.

Reply:
Can you type alias it?

`type X = *Y`

Reply:
Went with the alias route.